examples/training_scripts/nasbench201/nasbench201.py

```diff
@@ -17,156 +17,37 @@
-            from syne_tune.optimizer.schedulers.searchers.bayesopt.models.cost.linear_cost_model \
-                import NASBench201LinearCostModel
-
-            map_config_values = {
-                'skip_connect': NASBench201LinearCostModel.Op.SKIP_CONNECT,
-                'none': NASBench201LinearCostModel.Op.NONE,
-                'nor_conv_1x1': NASBench201LinearCostModel.Op.NOR_CONV_1x1,
-                'nor_conv_3x3': NASBench201LinearCostModel.Op.NOR_CONV_3x3,
-                'avg_pool_3x3': NASBench201LinearCostModel.Op.AVG_POOL_3x3,
-            }
-            conv_separate_features = ('cnvsep' in cost_model_type)
-            count_sum = ('sum' in cost_model_type)
-            cost_model = NASBench201LinearCostModel(
-                config_keys=CONFIG_KEYS,
-                map_config_values=map_config_values,
-                conv_separate_features=conv_separate_features,
-                count_sum=count_sum)
-        else:
-            from syne_tune.optimizer.schedulers.searchers.bayesopt.models.cost.sklearn_cost_model \
-                import ScikitLearnCostModel
-
-            cost_model = ScikitLearnCostModel(cost_model_type)
-        return cost_model
-    except Exception:
-        return None
-
-
-def download_datafile(dataset_path, dataset_name, dataset_s3_bucket):
-    assert dataset_name in ["ImageNet16-120", "cifar10-valid", "cifar100"]
-    assert dataset_s3_bucket is not None, \
-        "TODO: Need dataset_s3_bucket to point to bucket where nasbench201 data can be downloaded"
-    dataset_fname = f"nasbench201_reduced_{dataset_name}.csv"
-    s3_path = 'dataset'
-    fname_local = os.path.join(dataset_path, dataset_fname)
-    if not os.path.isfile(fname_local):
-        os.makedirs(dataset_path, exist_ok=True)
-        import boto3
-        s3 = boto3.resource('s3')
-        s3.meta.client.download_file(
-            dataset_s3_bucket, os.path.join(s3_path, dataset_fname), fname_local)
-    return fname_local
-
-
-def get_dataframe(dataset_path, dataset_name, dataset_s3_bucket):
-    from filelock import SoftFileLock, Timeout
-    import pandas
-
-    # Make sure the datafile is on the local filesystem
-    os.makedirs(dataset_path, exist_ok=True)
-    # Lock protection is needed for backends which run multiple worker
-    # processes on the same instance
-    lock_path = os.path.join(dataset_path, 'lock')
-    lock = SoftFileLock(lock_path)
-    try:
-        with lock.acquire(timeout=120, poll_intervall=1):
-            fname_local = download_datafile(
-                dataset_path, dataset_name, dataset_s3_bucket)
-    except Timeout:
-        print(
-            "WARNING: Could not obtain lock for dataset files. Trying anyway...",
-            flush=True)
-        fname_local = download_datafile(
-            dataset_path, dataset_name, dataset_s3_bucket)
-    return pandas.read_csv(fname_local)
+METRIC_ELAPSED_TIME = 'metric_elapsed_time'
 
 
 def objective(config):
```

Comment:
While I agree, to benchmark the simulator backend, it would be nice to have a version of the benchmark that allows us to sleep during the evaluation

Reply:
Assuming we need it, we could still have a code that is generic compare to having to write a main script that sleep for every benchmark (which can be done in another PR). The use-case is not 100% clear to me, is it to test the simulator to show that it gives the same result when you sleep or when you dont?

Reply:
Aaron nailed it. We'd like to be able to use (some) blackboxes also with the other back-ends.

Of course, this does not mean we have to wrap every blackbox in this way, just maybe a few ones that we'd like to use in this way. It is really optional.

Reply:
I agree this might be a bit specific and only relevant for a subset of the benchmarks, since it's only to benchmark the simulator backend.

Reply:
Good idea, David, to create a generic wrapper (it will be a little difficult, because each BB has its own config arguments).

For now, we'd need this in order to compare simulator against not (with sleep).

The case dont_sleep = True is indeed a bit obsolete now. Aaron and me wanted to run some experiments, though, we were do not sleep and use the local back-end, and show that this does not lead to the right results. This would be easier than (say) feeding a BB directly into the local back-end.

Reply:
Ok I fine not doing for now but is it something we can remove once we are not using anymore since supporting this adds a extra cases to maintain?